drivers/gpio/gpio_ite_it8xxx2_v2.c

```diff
@@ -0,0 +1,460 @@
+#define GPIO_GROUP_COUNT 13
+/* Keyboard scan group ksi, ksoh1 and ksol */
+#define KBS_GROUP_COUNT 3
+
+/**
+ * Driver functions
+ */
+static int gpio_ite_configure(const struct device *dev,
+			      gpio_pin_t pin,
+			      gpio_flags_t flags)
+{
+	const struct gpio_ite_cfg *gpio_config = dev->config;
+	volatile uint8_t *reg_gpdr = (uint8_t *)gpio_config->reg_gpdr;
+	volatile uint8_t *reg_gpotr = (uint8_t *)gpio_config->reg_gpotr;
+	volatile uint8_t *reg_p18scr = (uint8_t *)gpio_config->reg_p18scr;
+	volatile uint8_t *reg_gpcr = (uint8_t *)gpio_config->reg_gpcr + pin;
+	struct gpio_ite_data *data = dev->data;
+	uint8_t mask = BIT(pin);
+
+	__ASSERT(gpio_config->index < GPIO_GROUP_COUNT + KBS_GROUP_COUNT,
+		 "Invalid GPIO group index");
+
+	/* Don't support "open source" mode */
+	if (((flags & GPIO_SINGLE_ENDED) != 0) &&
+	    ((flags & GPIO_LINE_OPEN_DRAIN) == 0)) {
+		return -ENOTSUP;
+	}
+
+	if (flags == GPIO_DISCONNECTED) {
+		*reg_gpcr = GPCR_PORT_PIN_MODE_TRISTATE;
+		/*
+		 * Since not all GPIOs can be to configured as tri-state,
+		 * prompt error if pin doesn't support the flag.
+		 */
+		if (*reg_gpcr != GPCR_PORT_PIN_MODE_TRISTATE) {
+			/* Go back to default setting (input) */
+			*reg_gpcr = GPCR_PORT_PIN_MODE_INPUT;
+			LOG_ERR("Cannot config GPIO-%c%d as tri-state",
+				(gpio_config->index + 'A'), pin);
+			return -ENOTSUP;
+		}
+		/*
+		 * The following configuration isn't necessary because the pin
+		 * was configured as disconnected.
+		 */
+		return 0;
+	}
+
+	/*
+	 * Select open drain first, so that we don't glitch the signal
+	 * when changing the line to an output.
+	 */
+	if (flags & GPIO_OPEN_DRAIN) {
+		*reg_gpotr |= mask;
+	} else {
+		*reg_gpotr &= ~mask;
+	}
+
+	/* 1.8V or 3.3V */
+	if (gpio_config->has_volt_sel[pin]) {
+		gpio_flags_t volt = flags & IT8XXX2_GPIO_VOLTAGE_MASK;
+
+		if (volt == IT8XXX2_GPIO_VOLTAGE_1P8) {
+			__ASSERT(!(flags & GPIO_PULL_UP),
+			"Don't enable internal pullup if 1.8V voltage is used");
+			*reg_p18scr |= mask;
+			data->volt_default_set &= ~mask;
+		} else if (volt == IT8XXX2_GPIO_VOLTAGE_3P3) {
+			*reg_p18scr &= ~mask;
+			/*
+			 * A variable is needed to store the difference between
+			 * 3.3V and default so that the flag can be distinguished
+			 * between the two in gpio_ite_get_config.
+			 */
+			data->volt_default_set &= ~mask;
+		} else if (volt == IT8XXX2_GPIO_VOLTAGE_DEFAULT) {
+			*reg_p18scr &= ~mask;
+			data->volt_default_set |= mask;
+		} else {
+			return -EINVAL;
+		}
+	}
+
+	/* If output, set level before changing type to an output. */
+	if (flags & GPIO_OUTPUT) {
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			*reg_gpdr |= mask;
+		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
+			*reg_gpdr &= ~mask;
+		}
+	}
+
+	/* Set input or output. */
+	if (flags & GPIO_OUTPUT)
+		*reg_gpcr = (*reg_gpcr | GPCR_PORT_PIN_MODE_OUTPUT) &
+				~GPCR_PORT_PIN_MODE_INPUT;
+	else
+		*reg_gpcr = (*reg_gpcr | GPCR_PORT_PIN_MODE_INPUT) &
+				~GPCR_PORT_PIN_MODE_OUTPUT;
```

Comment:
Please add braces even for single-line if bodies.

Reply:
Done.